app/lib/surface/AnimateSprite.coffee

```diff
@@ -0,0 +1,331 @@
+createjs = require 'lib/createjs-parts'
+
+# Put this on MovieClips
+specialGoToAndStop = (frame) ->
+  if frame is @currentFrame and @childrenCopy
+    @addChild(@childrenCopy...)
+  else
+    @gotoAndStop(frame)
+    @childrenCopy = @children.slice(0)
+
+module.exports = class SegmentedSprite extends createjs.Container
```

Comment:
This is a fork of SegmentedSpite and is intended to represent our next generation rendering flow.  We can modify this as necessary without needing to worry about legacy flows.  For now only change is caching shapes.  It's commented below.